pkg/pipeline/simple_end_to_end_test.go

```diff
@@ -0,0 +1,103 @@
+/*
```

Comment:
would you want to just add this test as another test function under `pipeline_test.go`  ... I do not think that this is a complete end-to-end test ... this is more of end-to-end pipeline test.

Reply:
moved to pipeline_test.go